ext/lib/crypto/mbedtls/configs/config-tls-generic.h

```diff
@@ -0,0 +1,270 @@
+/*
+ * Copyright (C) 2006-2015, ARM Limited, All Rights Reserved
+ * Copyright (c) 2017 Intel Corporation.
+ * Copyright (c) 2018 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Generic configuration for TLS, manageable by Kconfig.
+ */
+
+#ifndef MBEDTLS_CONFIG_H
+#define MBEDTLS_CONFIG_H
+
+/* System support */
+#define MBEDTLS_PLATFORM_C
+#define MBEDTLS_PLATFORM_MEMORY
+#define MBEDTLS_MEMORY_BUFFER_ALLOC_C
+#define MBEDTLS_PLATFORM_NO_STD_FUNCTIONS
+#define MBEDTLS_PLATFORM_EXIT_ALT
+#define MBEDTLS_NO_PLATFORM_ENTROPY
+#define MBEDTLS_NO_DEFAULT_ENTROPY_SOURCES
+#define MBEDTLS_PLATFORM_PRINTF_ALT
+
+#if !defined(CONFIG_ARM)
+#define MBEDTLS_HAVE_ASM
+#endif
+
+#if defined(CONFIG_MBEDTLS_TEST)
+#define MBEDTLS_SELF_TEST
+#define MBEDTLS_DEBUG_C
+#endif
+
+/* mbedTLS feature support */
+
+/* Supported TLS versions */
+#if defined(CONFIG_NET_TLS)
+#if defined(CONFIG_TLS_VERSION_1_0)
+#define MBEDTLS_SSL_PROTO_TLS1
+#endif
+
+#if defined(CONFIG_TLS_VERSION_1_1)
+#define MBEDTLS_SSL_PROTO_TLS1_1
+#endif
+
+#if defined(CONFIG_TLS_VERSION_1_2)
+#define MBEDTLS_SSL_PROTO_TLS1_2
+#endif
+
+/* Module required for TLS */
+#define MBEDTLS_SSL_TLS_C
+#define MBEDTLS_SSL_SRV_C
+#define MBEDTLS_SSL_CLI_C
+#define MBEDTLS_CIPHER_C
+#define MBEDTLS_MD_C
+
+/* TODO Perhaps make this configurable? */
+#define MBEDTLS_SSL_MAX_FRAGMENT_LENGTH
```

Comment:
I guess you refer to `MBEDTLS_SSL_MAX_CONTENT_LEN` here? `MBEDTLS_SSL_MAX_FRAGMENT_LENGTH` is just hardcoded in some of the config files.

Reply:
Yes, that's what I refer to.

> MBEDTLS_SSL_MAX_FRAGMENT_LENGTH is just hardcoded in some of the config files.

That's because there're too many config files, and a bit hard to update them consistently (without risking regressions). However, as the linked PR shows, there's a Kconfig option in Zephyr to set TLS fragment size, and all newer configs should use it.

Reply:
These are two different configs, though they look pretty similar:
`MBEDTLS_SSL_MAX_FRAGMENT_LENGTH `
`MBEDTLS_SSL_MAX_CONTENT_LEN`

 `TODO` in the code refers to `MBEDTLS_SSL_MAX_FRAGMENT_LENGTH`, which enables `max_fragment_length` TLS extension, I saw it was present in some of the tls-related config files, yet I've left a remainder to investigate the use-case of it in future.

`MBEDTLS_SSL_MAX_CONTENT_LEN` is configured with the option from Kconfig, just as in the PR posted (at the very bottom of `config-tls-generic.h` file).

Reply:
D'oh, sorry for this noise, indeed, I mixed up these two options :-(

Regarding fragment length extension, I'd say it makes sense to enable it by default (like it's done), yet there're too few real-world sites with which it works :-(.

Reply:
> yet there're too few real-world sites with which it works :-(.

Yep, I've faced that already :/